core/services/log/broadcaster_test.go

```diff
@@ -520,6 +520,93 @@ func TestBroadcaster_BroadcastsAtCorrectHeights(t *testing.T) {
 	helper.mockEth.assertExpectations(t)
 }
 
+func TestBroadcaster_BroadcastsWithZeroConfirmations(t *testing.T) {
+	t.Parallel()
+
+	const blockHeight int64 = 10
+	helper := newBroadcasterHelper(t, blockHeight, 1)
+	helper.store.Config.Set(config.EnvVarName("EthFinalityDepth"), uint(1))
+	helper.start()
+
+	contract1, err := flux_aggregator_wrapper.NewFluxAggregator(cltest.NewAddress(), nil)
+	require.NoError(t, err)
+
+	blocks := cltest.NewBlocks(t, 10)
+	addr1SentLogs := []types.Log{
+		cltest.RawNewRoundLog(t, contract1.Address(), utils.NewHash(), 100, 0, false),
+		cltest.RawNewRoundLog(t, contract1.Address(), utils.NewHash(), 101, 0, false),
+		cltest.RawNewRoundLog(t, contract1.Address(), utils.NewHash(), 105, 0, false),
+	}
+
+	listener1 := helper.newLogListener("listener 1")
+	listener2 := helper.newLogListener("listener 2")
+
+	helper.register(listener1, contract1, 0)
+	helper.register(listener2, contract1, 0)
+
+	cleanup, _ := cltest.SimulateIncomingHeads(t, cltest.SimulateIncomingHeadsArgs{
+		StartBlock:     0,
+		EndBlock:       9,
+		BackfillDepth:  10,
+		HeadTrackables: []httypes.HeadTrackable{(helper.lb).(httypes.HeadTrackable)},
+		Blocks:         blocks,
+		Interval:       250 * time.Millisecond,
+	})
+	defer cleanup()
+
+	chRawLogs := <-helper.chchRawLogs
+
+	for _, log := range addr1SentLogs {
+		chRawLogs <- log
+	}
+
+	requireBroadcastCount(t, helper.store, 6)
```

Comment:
I don't love this requireBroadcastCount as it leans on gomegas consistently + eventually. Could we just install a buffered channel in the helper which HandleLog writes to and we just read from 6 times here (or timeout) ?

Reply:
do we want to reduce the usage of gomega? i thought it's quite handy - e.g. compared to `require.Eventually` (e.g. `require.Eventually(t, func() bool { return len(idleDurationOccured) == 2 }, 3*time.Second, 10*time.Millisecond)` ) it prints the actual value at the timeout, if it's as expected

Reply:
but yes a more direct check of sends will be nice, rather than going via the DB

Reply:
hm though it's a bit tricky - currently simpleLogListener does not have a reference to the helper (where i'd put the channel) - so i'm rather tempted to do it in another PR